### PR TITLE
Base container

### PR DESCRIPTION
**Pull request by drivera-armedia**
First version of the container to be merged to the main branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,24 @@
+ARG ARCH="amd64"
+ARG OS="linux"
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+LABEL maintainer="Armedia, LLC"
 
+ARG ARCH="amd64"
```
Comment (fabricetriboix) [inline Dockerfile:6]: Duplicate of line 1

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,24 @@
...
+ARG OS="linux"
```
Comment (fabricetriboix) [inline Dockerfile:7]: Duplicate of line 2

File: Dockerfile [hunk 3]
```diff
@@ -1 +1,24 @@
...
+ARG VER="1.2.0"
+ARG PKG="node_exporter"
+ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
+
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
+		-o package.tar.gz && \
```
Comment (fabricetriboix) [inline Dockerfile]: Same comment about S3 bucket and `get-artifcats.sh` as in the `ark_prometheus` PR

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,24 @@
...
+	tar -xzvf "package.tar.gz" && \
+	mv -vif "${SRC}/LICENSE"                     "/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"                      "/NOTICE" && \
+	mv -vif "${SRC}/node_exporter"               "/bin/node_exporter" && \
+	chown nobody:nobody "/bin/node_exporter" && \
+	rm -rvf "${SRC}" package.tar.gz
+
+USER        nobody
```
Comment (fabricetriboix) [inline Dockerfile]: Please create a user using `useradd` and do not use the `nobody` user

File: Dockerfile [hunk 5]
```diff
@@ -1 +1,24 @@
...
+ARG SRC="${PKG}-${VER}.${OS}-${ARCH}"
+
+RUN curl \
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
+		-o package.tar.gz && \
+	tar -xzvf "package.tar.gz" && \
+	mv -vif "${SRC}/LICENSE"                     "/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"                      "/NOTICE" && \
+	mv -vif "${SRC}/node_exporter"               "/bin/node_exporter" && \
+	chown nobody:nobody "/bin/node_exporter" && \
```
Comment (fabricetriboix) [inline Dockerfile]: Please change this to `root:root`

**Review by fabricetriboix — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 6]
```diff
@@ -1 +1,27 @@
...
+FROM 345280441424.dkr.ecr.ap-south-1.amazonaws.com/ark_base:latest
+
+LABEL	ORG="Armedia LLC" \
+		APP="Prometheus Node Exporter" \
+		VERSION="${VER}" \
+		IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_nodeexp" \
+		MAINTAINER="Armedia LLC"
+
+# Modify to fetch from S3 ...
+RUN curl \
```
Comment (JonHolman) [inline Dockerfile:30]: add get-artifacts script, put this in our artifacts bucket

File: Dockerfile [hunk 7]
```diff
@@ -1 +1,27 @@
...
+		-L "https://github.com/prometheus/${PKG}/releases/download/v${VER}/${SRC}.tar.gz" \
+		-o "package.tar.gz" && \
+	tar -xzvf "package.tar.gz" && \
+	mv -vif "${SRC}/LICENSE"					"/LICENSE" && \
+	mv -vif "${SRC}/NOTICE"						"/NOTICE" && \
+	mv -vif "${SRC}/node_exporter"				"/bin/node_exporter" && \
+	rm -rvf "${SRC}" "package.tar.gz"
+
+USER		nobody
```
Comment (JonHolman) [inline Dockerfile]: As @fabricetriboix said on the other PR, please don't use nobody.  Create a user and use it.

File: Dockerfile [hunk 8]
```diff
@@ -1 +1,27 @@
...
+EXPOSE		9100
+ENTRYPOINT	[ "/bin/node_exporter" ]
```
Comment (JonHolman) [inline Dockerfile]: as @fabricetriboix said on the other PR, please don't use ENTRYPOINT, use CMD

**Review by JonHolman — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 9]
```diff
@@ -1,27 +1,37 @@
...
-LABEL	ORG="Armedia LLC" \
-		APP="Prometheus Node Exporter" \
-		VERSION="${VER}" \
-		IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_nodeexp" \
-		MAINTAINER="Armedia LLC"
+LABEL   ORG="Armedia LLC" \
+        APP="Prometheus Node Exporter" \
+        VERSION="${VER}" \
+        IMAGE_SOURCE="https://github.com/ArkCase/ark_prometheus_nodeexp" \
+        MAINTAINER="Armedia LLC"
```
Comment (fabricetriboix) [inline Dockerfile]: Not consistent with ark_prometheus

**Review by fabricetriboix — DISMISSED** [review]

**Review by fabricetriboix — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Review by JonHolman — APPROVED** [review]

**Pull request merged**